### PR TITLE
doc/rados: edit crush-map.rst (2 of x)

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/rados/operations/crush-map.rst.

https://tracker.ceph.com/issues/58485

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/operations/crush-map.rst [hunk 1]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
 .. _addosd:
 
-Add/Move an OSD
----------------
+Adding/Moving an OSD
+--------------------
 
-.. note: OSDs are normally automatically added to the CRUSH map when
-         the OSD is created.  This command is rarely needed.
+.. note:: Normally, OSDs automatically add themselves to the CRUSH map when
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:337]: Two adverbs seems like overkill, I would s/Normally, //

File: doc/rados/operations/crush-map.rst [hunk 2]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
+   :Example: ``2.0``
 
 
 ``root``
-
-:Description: The root node of the tree in which the OSD resides (normally ``default``)
-:Type: Key/value pair.
-:Required: Yes
-:Example: ``root=default``
+   :Description: The root node of the tree in which the OSD resides (normally ``default``).
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:365]: s/tree/CRUSH hierarchy/

File: doc/rados/operations/crush-map.rst [hunk 3]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
 
-Adjust OSD weight
------------------
+Adjusting OSD weight
+--------------------
 
-.. note: Normally OSDs automatically add themselves to the CRUSH map
-         with the correct weight when they are created. This command
-         is rarely needed.
+.. note:: Normally OSDs automatically add themselves to the CRUSH map with the
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:388]: s/Normally//

File: doc/rados/operations/crush-map.rst [hunk 4]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
+   correct weight when they are created. The command in this section is rarely
+   needed.
 
-To adjust an OSD's CRUSH weight in the CRUSH map of a running cluster, execute
-the following:
+To adjust an OSD's CRUSH weight in the CRUSH map of a running cluster, run a
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:392]: s/the CRUSH map of//

File: doc/rados/operations/crush-map.rst [hunk 5]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
 .. _removeosd:
 
-Remove an OSD
--------------
+Removing an OSD
+---------------
 
-.. note: OSDs are normally removed from the CRUSH as part of the
-   ``ceph osd purge`` command.  This command is rarely needed.
+.. note:: OSDs are normally removed from the CRUSH as a result of the ``ceph
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:420]: the CRUSH map

File: doc/rados/operations/crush-map.rst [hunk 6]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
-.. note: Buckets are implicitly created when an OSD is added
-   that specifies a ``{bucket-type}={bucket-name}`` as part of its
-   location,  if a bucket with that name does not already exist.  This
-   command is typically used when manually adjusting the structure of the
-   hierarchy after OSDs have been created.  One use is to move a
-   series of hosts underneath a new rack-level bucket; another is to
-   add new ``host`` buckets (OSD nodes) to a dummy ``root`` so that they don't
-   receive data until you're ready, at which time you would move them to the
-   ``default`` or other root as described below.
+Adding a Bucket
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:439]: Suggest "Adding a CRUSH bucket" to disambiguate from S3, though anyone who thinks we're talking about S3 here is in the wrong place.

File: doc/rados/operations/crush-map.rst [hunk 7]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
-:Required: Yes
-:Example: ``rack12``
+   :Description: The full name of the bucket.
+   :Type: String
+   :Required: Yes
+   :Example: ``rack12``
 
 
 ``bucket-type``
+   :Description: The type of the bucket. This type must already exist in the hierarchy.
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:470]: CRUSH hierarchy

File: doc/rados/operations/crush-map.rst [hunk 8]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
-Where:
+For details on this command's parameters, see the following:
 
 ``bucket-name``
-
-:Description: The name of the bucket to move/reposition.
-:Type: String
-:Required: Yes
-:Example: ``foo-bar-1``
+   :Description: The name of the bucket that you are moving or repositioning.
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:494]: s/or repositioning//

synonyms?

File: doc/rados/operations/crush-map.rst [hunk 9]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
+   :Type: String
+   :Required: Yes
+   :Example: ``foo-bar-1``
 
 ``bucket-type``
+   :Description: The bucket's location in the CRUSH hierarchy.
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:500]: new location

File: doc/rados/operations/crush-map.rst [hunk 10]
```diff
@@ -326,214 +326,204 @@ Although weight sets can be set up and adjusted manually, we recommend enabling
...
-To remove a bucket from the CRUSH hierarchy, execute the following:
+To remove a bucket from the CRUSH hierarchy, run a command of the following
+form:
 
 .. prompt:: bash $
 
    ceph osd crush remove {bucket-name}
 
-.. note:: A bucket must be empty before removing it from the CRUSH hierarchy.
+.. note:: A bucket must already be empty before it is removed from the CRUSH hierarchy.
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:515]: Suggest adding:

In other words, there must not be OSDs or any other CRUSH buckets within it.

File: doc/rados/operations/crush-map.rst [hunk 11]
```diff
@@ -542,22 +532,24 @@ The following example removes the ``rack12`` bucket from the hierarchy:
 Creating a compat weight set
 ----------------------------
 
-.. note: This step is normally done automatically by the ``balancer``
-   module when enabled.
+.. note:: Normally this action is done automatically by the ``balancer`` module
```
Comment (anthonyeleven) [inline doc/rados/operations/crush-map.rst:535]: automatically if needed by the

Few people should be using the crush-compat mode with the most recent releases.

**Review by anthonyeleven — APPROVED** [review]

File: doc/rados/operations/crush-map.rst (hunk 1 above)
Comment (colemitchell) [reply doc/rados/operations/crush-map.rst:337]: @zdover23 - Maybe solve the adverb overexposure by replacing _Normally_ with _Under normal conditions_.

Comment (zdover23) [reply doc/rados/operations/crush-map.rst:337]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:365]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:388]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 4 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:392]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 5 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:420]: Accepted.

Good catch.

File: doc/rados/operations/crush-map.rst (hunk 6 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:439]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 7 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:470]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 8 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:494]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 9 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:500]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 10 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:515]: Accepted.

File: doc/rados/operations/crush-map.rst (hunk 11 above)
Comment (zdover23) [reply doc/rados/operations/crush-map.rst:535]: Accepted.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/52069 - Reef backport
https://github.com/ceph/ceph/pull/52070 - Quincy backport

**Pull request merged**